# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/hotfix-armeb.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- firefox-107.0/build/moz.configure/init.configure.vanilla	2022-11-17 17:55:31.419221937 +0100
+++ firefox-107.0/build/moz.configure/init.configure	2022-11-17 17:57:55.802208026 +0100
@@ -529,7 +529,7 @@
         endianness = "big"
     elif cpu.startswith("arm"):
         canonical_cpu = "arm"
-        endianness = "big" if cpu.startswith(("armeb", "armbe")) else "little"
+        endianness = "big" if cpu.endswith(("eb", "be")) else "little"
     elif cpu in ("m68k"):
         canonical_cpu = "m68k"
         endianness = "big"
--- firefox-107.0/third_party/libwebrtc/build/build_config.h.vanilla	2022-11-17 18:02:57.834178924 +0100
+++ firefox-107.0/third_party/libwebrtc/build/build_config.h	2022-11-17 18:03:29.928175831 +0100
@@ -165,6 +165,11 @@
 #define ARCH_CPU_ARMEL 1
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_LITTLE_ENDIAN 1
+#elif defined(__ARMEB__)  
+#define ARCH_CPU_ARM_FAMILY 1
+#define ARCH_CPU_ARMEL 1
+#define ARCH_CPU_32_BITS 1
+#define ARCH_CPU_BIG_ENDIAN 1
 #elif defined(__aarch64__) || defined(_M_ARM64)
 #define ARCH_CPU_ARM_FAMILY 1
 #define ARCH_CPU_ARM64 1
